codex/stores/repostore.nim

```diff
@@ -97,7 +102,53 @@ method getBlock*(self: RepoStore, cid: Cid): Future[?!Block] {.async.} =
     return failure(newException(BlockNotFoundError, err.msg))
 
   trace "Got block for cid", cid
-  return Block.new(cid, data)
+  return Block.new(cid, data, verify = true)
+
+proc getMerkleTree(self: RepoStore, cid: Cid): Future[?!MerkleTree] {.async.} =
+  try:
```

Comment:
I'd suggest avoiding exceptions (specially mixing wiht result), you can either use `.catch` from `questionable` (?) or the `withValue` apis (this are ugly but performant).

Reply:
Fixed in #566 , I'm using `.getOption` on `LruCache` instead